### ISSUE TITLE
False positive "no-member" for Python 3 subprocess.Popen().communicate()

### ISSUE THREAD

**Issue by pylint-bot** [PyCQA/pylint#64]
Originally reported by: **Anonymous**

---

Pylint assumes that Popen().communicate() returns a tuple of 2 "str" instance, but this is incorrect in Python 3 where it returns a tuple of 2 "bytes" instances, unless universal_newlines=True is passed to Popen, in which case it returns a tuple of 2 "str" instances again.

$ cat subprocess_test.py
from subprocess import Popen, PIPE
stdout, stderr = Popen('/bin/ls', stdout=PIPE).communicate()
print(stdout.decode("ascii"))

$ pylint -d C -r n subprocess_test.py
************\* Module subprocess_test
E:  3, 6: Instance of 'str' has no 'decode' member (no-member)

Python 3 runs the code fine.

I wonder if this is related to the "brain" package included in asteroid, as in the asteroid package:

$ grep -A 12 Popen brain/py2stdlib.py 
class Popen(object):
    returncode = pid = 0
    stdin = stdout = stderr = file()

```
def __init__(self, args, bufsize=0, executable=None,
             stdin=None, stdout=None, stderr=None,
             preexec_fn=None, close_fds=False, shell=False,
             cwd=None, env=None, universal_newlines=False,
             startupinfo=None, creationflags=0):
    pass

def communicate(self, input=None):
    return ('string', 'string')
```

---
- Bitbucket: https://bitbucket.org/logilab/pylint/issue/65
